torch/_prims/rng_prims.py

```diff
@@ -62,7 +62,7 @@ def philox_rand_offset(
     numel_scalar = 1
     for dim_size in shape:
         numel_scalar *= dim_size
-    numel = torch.scalar_tensor(numel_scalar, dtype=torch.int64)
+    numel = cast(float, torch.scalar_tensor(numel_scalar, dtype=torch.int64))
```

Comment:
this cast doesn't look safe, the scalar tensor is clearly an int64? Surely it's better to use a to(torch.float32) or a `.to(torch.float32)` item()? Using torch.floor or round is probably better though? I don't like how we are lying about the type here with a cast.